test/general/key.js

```diff
@@ -3934,4 +4028,48 @@ XvmoLueOOShu01X/kaylMqaT8w==
     expect(signingKeySignature instanceof openpgp.SignaturePacket).to.be.true;
     expect(signingKeySignature.keyFlags[0] & openpgp.enums.keyFlags.signData).to.be.equals(openpgp.enums.keyFlags.signData);
   });
+
+  it("Sign and verify key - key with user attribute", async function () {
+    let publicKey = await openpgp.readKey({ armoredKey: keyWithImageData });
+    const privateKey = await openpgp.decryptKey({
+      privateKey: await openpgp.readKey({ armoredKey: priv_key_rsa }),
+      passphrase: "hello world",
+    });
+
+    const { minRSABits } = openpgp.config;
+    openpgp.config.minRSABits = 1024;
+    try {
+      publicKey = await publicKey.signAllUsers([privateKey]);
+      const signatures = await publicKey.verifyAllUsers([privateKey]);
+      const publicSigningKey = await publicKey.getSigningKey();
+      const privateSigningKey = await privateKey.getSigningKey();
+      expect(signatures.length).to.equal(4);
+      expect(signatures[0].userID).to.equal(publicKey.users[0].userID.userID);
+      expect(signatures[0].userAttribute).to.be.null;
+      expect(signatures[0].keyID.toHex()).to.equal(
+        publicSigningKey.getKeyID().toHex()
+      );
+      expect(signatures[0].valid).to.be.null;
+      expect(signatures[1].userID).to.equal(publicKey.users[0].userID.userID);
+      expect(signatures[1].userAttribute).to.be.null;
+      expect(signatures[1].keyID.toHex()).to.equal(
+        privateSigningKey.getKeyID().toHex()
+      );
+      expect(signatures[1].valid).to.be.true;
+      expect(signatures[2].userID).to.be.null;
+      expect(signatures[2].userAttribute.attributes[0]).to.be.not.empty;
+      expect(signatures[2].keyID.toHex()).to.equal(
+        publicSigningKey.getKeyID().toHex()
+      );
+      expect(signatures[2].valid).to.be.null;
+      expect(signatures[3].userID).to.be.null;
+      expect(signatures[3].userAttribute.attributes[0]).to.be.not.empty;
+      expect(signatures[3].keyID.toHex()).to.equal(
+        privateSigningKey.getKeyID().toHex()
+      );
+      expect(signatures[3].valid).to.be.true;
+    } finally {
+      openpgp.config.minRSABits = minRSABits;
+    }
+  });
 });
```

Comment:
It's not required but if you still have time & energy, do you want to add a test for the case of verifying with a wrong key (like [this one](https://github.com/openpgpjs/openpgpjs/blob/726ee55de41a2ca305fce4c12fcd24696869bbc4/test/general/key.js#L2523-L2554)) too?

Reply:
yes, no problem. thank you!